lib/helpers/emitter.js

```diff
@@ -20,28 +20,34 @@ class Logger {
   }
 }
 
+class EmitterError extends Error { }
+exports.EmitterError = EmitterError;
+
 exports.Emitter = class Emitter extends EventEmitter {
   constructor() {
     super(EventEmitter);
     EventEmitter.call(this);
     this.logger = new Logger();
+    this.VALID_EVENTS = new Set(['data', 'error', 'snapshot', 'end']);
```

Comment:
It might be worth adding the rest of the valid events in this PR?

```
    scope
        .on('data', onData)
        .on('error', onError)
        .on('rebound', onRebound)
        .on('snapshot', onSnapshot)
        .on('updateSnapshot', onUpdateSnapshot)
        .on('updateKeys', onUpdateKeys)
        .on('httpReply', onHttpReply)
        .on('end', onEnd);
```

These ones are all valid in code according to Sailor